From 123eec2c286ab70444d8e0b1dd50e48429b350af Mon Sep 17 00:00:00 2001
From: RH udev team <udev-maint@redhat.com>
Date: Fri, 26 Jul 2013 15:26:48 +0200
Subject: [PATCH] cdrom_id: remove unused var and option

---
 extras/cdrom_id/cdrom_id.c | 3 ---
 1 file changed, 3 deletions(-)

diff --git a/extras/cdrom_id/cdrom_id.c b/extras/cdrom_id/cdrom_id.c
index 2ed867a..d8971fd 100644
--- a/extras/cdrom_id/cdrom_id.c
+++ b/extras/cdrom_id/cdrom_id.c
@@ -835,7 +835,6 @@ int main(int argc, char *argv[])
 		{}
 	};
 	const char *node = NULL;
-	int export = 0;
 	int fd = -1;
 	int cnt;
 	int rc = 0;
@@ -861,11 +860,9 @@ int main(int argc, char *argv[])
 				udev_set_log_priority(udev, LOG_INFO);
 			break;
 		case 'x':
-			export = 1;
 			break;
 		case 'h':
 			printf("Usage: cdrom_id [options] <device>\n"
-			       "  --export        export key/value pairs\n"
 			       "  --debug         debug to stderr\n"
 			       "  --help          print this help text\n\n");
 			goto exit;
-- 
1.8.3.1

