diff -up udev-147/rules/rules.d/60-persistent-storage.rules.pciessd udev-147/rules/rules.d/60-persistent-storage.rules
--- udev-147/rules/rules.d/60-persistent-storage.rules.pciessd	2014-06-06 14:25:20.989595435 +0200
+++ udev-147/rules/rules.d/60-persistent-storage.rules	2014-06-06 14:27:20.392598280 +0200
@@ -27,6 +27,11 @@ KERNEL=="hd*[!0-9]", IMPORT{program}="at
 KERNEL=="hd*[!0-9]", ENV{ID_SERIAL}=="?*", SYMLINK+="disk/by-id/ata-$env{ID_SERIAL}"
 KERNEL=="hd*[0-9]", ENV{ID_SERIAL}=="?*", SYMLINK+="disk/by-id/ata-$env{ID_SERIAL}-part%n"
 
+# by-id (hardware serial number) for Micron PCIe SSDs
+KERNEL=="rssd*[!0-9]", IMPORT{program}="ata_id --export $tempnode"
+KERNEL=="rssd*[!0-9]", ENV{ID_SERIAL_SHORT}=="?*", SYMLINK+="disk/by-id/ata-$env{ID_SERIAL_SHORT}"
+KERNEL=="rssd*[0-9]", ENV{ID_SERIAL_SHORT}=="?*", SYMLINK+="disk/by-id/ata-$env{ID_SERIAL_SHORT}-part%n"
+
 # virtio-blk
 KERNEL=="vd*[!0-9]", ATTRS{serial}=="?*", ENV{ID_SERIAL}="$attr{serial}", SYMLINK+="disk/by-id/virtio-$env{ID_SERIAL}"
 KERNEL=="vd*[0-9]", ATTRS{serial}=="?*", ENV{ID_SERIAL}="$attr{serial}", SYMLINK+="disk/by-id/virtio-$env{ID_SERIAL}-part%n"
