diff -up udev-147/extras/udev-acl/70-acl.rules.git500 udev-147/extras/udev-acl/70-acl.rules
--- udev-147/extras/udev-acl/70-acl.rules.git500	2011-02-03 11:02:47.000000000 +0100
+++ udev-147/extras/udev-acl/70-acl.rules	2009-08-14 20:13:13.000000000 +0200
@@ -49,7 +49,7 @@ ENV{ID_SMARTCARD_READER}=="*?", ENV{ACL_
 SUBSYSTEM=="input", ENV{ID_CLASS}=="joystick", ENV{ACL_MANAGE}="1"
 
 # apply ACL for all locally logged in users
-LABEL="acl_apply", ENV{ACL_MANAGE}=="?*", TEST=="/var/run/ConsoleKit/database", \
+LABEL="acl_apply", ENV{ACL_MANAGE}=="1", TEST=="/var/run/ConsoleKit/database", \
   RUN+="udev-acl --action=$env{ACTION} --device=$env{DEVNAME}"
 
 LABEL="acl_end"
diff -up udev-147/extras/udev-acl/udev-acl.c.git500 udev-147/extras/udev-acl/udev-acl.c
--- udev-147/extras/udev-acl/udev-acl.c.git500	2009-09-27 02:42:54.000000000 +0200
+++ udev-147/extras/udev-acl/udev-acl.c	2011-02-03 11:01:21.000000000 +0100
@@ -289,7 +289,7 @@ static void apply_acl_to_devices(uid_t u
 	/* iterate over all devices tagged with ACL_SET */
 	udev = udev_new();
 	enumerate = udev_enumerate_new(udev);
-	udev_enumerate_add_match_property(enumerate, "ACL_MANAGE", "*");
+	udev_enumerate_add_match_property(enumerate, "ACL_MANAGE", "1");
 	udev_enumerate_scan_devices(enumerate);
 	udev_list_entry_foreach(list_entry, udev_enumerate_get_list_entry(enumerate)) {
 		struct udev_device *device;
